### ISSUE TITLE
possible regression from v1.15.0

### ISSUE THREAD

**Issue by tmc** [jhump/protoreflect#553]
The [api-linter](https://github.com/googleapis/api-linter) tool starts failing for me if i update protoreflect to v1.15.0 with messages like:

```
2023/03/01 15:35:55 interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *
impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *im
pl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl.listReflect; interface conversion: protoreflect.List is *dynamicpb.dynamicList, not *impl
.listReflect
``` 

See [here](https://github.com/tmc/api-linter/actions/runs/4309032711/jobs/7515947958) for a sample run attempting to update this library.

**Comment by jhump** [comment]
@tmc, do you have the ability to test an unreleased commit? I think this regression should now be fixed, as of #554. Can you confirm? If you can, and it is indeed fixed, I will cut a v1.15.1.

**Comment by tmc** [comment]
> @tmc, do you have the ability to test an unreleased commit? I think this regression should now be fixed, as of #554. Can you confirm? If you can, and it is indeed fixed, I can cut a v1.15.1 ASAP.

Yes certainly. Will test now.

**Comment by tmc** [comment]
7923b758f8db6fe15447be78707e270e4920fa15 does fix this issue in google's api-linter

**Comment by jhump** [comment]
@tmc, thanks for testing. I will make a release later today.

**Comment by jhump** [comment]
Fixed in v1.15.1.
